DEVELOPMENT.md

```diff
@@ -5,51 +5,52 @@ The `Datadog/datadog-cloudformation-resources` repository contains:
+  * On failure, a handler should return an error message. A success does not return a message. For example:
 
-    ```
-    git clone git@github.com:DataDog/datadog-api-client-java.git
-    cd datadog-api-client-java
-    # This installs the client into ~/.m2/repository
-    mvn install -Dmaven.test.skip=true
-    ```
-2. Build `datadog-cloudformation-common`:
+   ```
+    return ProgressEvent(
+        status=OperationStatus.FAILED,
+        resourceModel=model,
+        message=f"Error getting monitor: 12345",
+        errorCode=http_to_handler_error_code(e.status)
+    )
+   ```
 
-    ```
-    # This installs the common package into ~/.m2/repository
-    mvn -f datadog-cloudformation-common/pom.xml -Dmaven.test.skip=true install
-    ```
-3. Install [`cfn-cli`](https://github.com/aws-cloudformation/cloudformation-cli).
+* Primary Identifiers should all be based on required fields. Having optional fields make up this property causes errors on stack creation. These will also be displayed when `Fn:Ref` is called on this resource.
+* Using the built in `logger` in the resource displays logs in CloudWatch to help debug any issues.
 
-## Testing
+### Testing the resource manually in AWS
 
-1. Follow the steps in [Setup](#setup).
+1. Install the prerequisites in [Prerequisites](#Prerequisites:).
 2. `cd` into the directory of the resource to be tested.
-3.  Run `mvn test` inside the directory to run the test suite for that resource.
+3. Run `cfn generate` to generate code based on the project and resource type schema.
+4. Run `cfn submit --set-default` to register the extension with CloudFormation and set it as the default version.
+5. Create a stack in AWS with the submitted resource.
 
-**Note**: the tests use `DD_TEST_CF_API_KEY`, `DD_TEST_CF_APP_KEY` and optionally also `DD_TEST_CF_API_URL` from environment variables.
 
-### Development Tips
+### Local testing using contract tests
 
-* The `Create` and `Update` handlers of your resource should call the `Read` handler (when the create/update is successful) to return a fully populated model.
-* On failure, a handler should return an error message. A success does not return a message. For example:
+AWS Cloudformation CLI provides tests to ensure resource type is behaving as expected during each event in the resource lifecycle.
+AWS requires all contracts tests to pass for resource to be registered.
 
-    ```
-    return ProgressEvent.<ResourceModel, CallbackContext>builder()
-        .resourceModel(model)
-        .status(OperationStatus.FAILED)
-        .message("Failed to read monitor 12345")
-        .build();
-    ```
+1. Install the prerequisites in [Prerequisites](#Prerequisites:).
+2. `cd` into the directory of the resource to be tested.
+3. Run `cfn generate` to generate code based on the project and resource type schema.
+4. Run `cfn submit --dry-run`. This will generate a zip file and place it in your current directory.
```

Comment:
```suggestion
4. Run `cfn submit --no-role --dry-run`. This will generate a zip file and place it in your current directory.
```
Can we omit the `--no-role` here ?

Reply:
I dont think we need to specify it here